packages/amplify-provider-awscloudformation/src/admin-modelgen.ts

```diff
@@ -0,0 +1,103 @@
+import ora from 'ora';
+import { $TSContext, stateManager, pathManager, $TSAny } from 'amplify-cli-core';
+import { isDataStoreEnabled } from 'graphql-transformer-core';
+import * as path from 'path';
+import { ProviderName as providerName } from './constants';
+import { isAmplifyAdminApp } from './utils/admin-helpers';
+import { AmplifyBackend } from './aws-utils/aws-amplify-backend';
+
+export async function adminModelgen(context: $TSContext, resources: $TSAny[]) {
+  resources = resources.filter(resource => resource.service === 'AppSync');
+
+  if (resources.length > 0) {
+    const resource = resources[0];
+    const { resourceName } = resource;
+
+    const amplifyMeta = stateManager.getMeta();
+    const localEnvInfo = stateManager.getLocalEnvInfo();
+
+    const appId = amplifyMeta.providers[providerName].AmplifyAppId;
+    const envName = localEnvInfo.envName;
+
+    const { isAdminApp } = await isAmplifyAdminApp(appId);
+    const isDSEnabled = await isDataStoreEnabled(path.join(pathManager.getBackendDirPath(), 'api', resourceName));
+
+    if (isAdminApp && isDSEnabled) {
+      // Generate DataStore Models for Admin UI CMS to consume
+      const spinner = ora('Generating models in the cloud…\n').start();
+      const amplifyBackendInstance = await new AmplifyBackend(context);
+      try {
+        const jobStartDetails = await amplifyBackendInstance.amplifyBackend
+          .generateBackendAPIModels({
+            AppId: appId,
+            BackendEnvironmentName: envName,
+            ResourceName: resourceName,
+          })
+          .promise();
+
+        const jobCompletionDetails = await pollUntilDone(
+          jobStartDetails.JobId,
+          appId,
+          envName,
+          2 * 1000,
+          2000 * 1000,
+          amplifyBackendInstance.amplifyBackend,
+        );
+        if (jobCompletionDetails.Status === 'COMPLETED') {
+          spinner.succeed('Successfully generated models in the cloud.');
+        } else {
+          throw new Error('Modelgen job creation failed');
+        }
+      } catch (e) {
+        spinner.stop();
+        console.log(e.stack);
+        context.print.error(`Failed to create models in the cloud`);
```

Comment:
should this throw instead of print? Seems like that would be more consistent with the error handling above